File: redis/redis-example/src/main/java/com/example/redisexample/controller/TestController.java
package com.example.redisexample.controller;


import com.example.redisexample.dto.PointDto;
import com.example.redisexample.repository.PointRedisRepository;
import com.example.redisexample.service.PointRedisService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class TestController {

    private final PointRedisService pointRedisService;
    private final PointRedisRepository pointRedisRepository;

    @GetMapping("/point")
    public PointDto visit(Long id) {
        return pointRedisService.savePoint(id);
    }

    @GetMapping("/list")
    public List<PointDto> getAll() {
        return (List<PointDto>) pointRedisRepository.findAll();
    }
}

File: r2db-mysql-example/src/main/kotlin/com/example/r2dbmysqlexample/controller/TestController.kt
package com.example.r2dbmysqlexample.controller

import com.example.r2dbmysqlexample.config.KLogger
import com.example.r2dbmysqlexample.persistence.entity.TestTlb
import com.example.r2dbmysqlexample.persistence.respository.TestTlbRepository
import kotlinx.coroutines.delay
import kotlinx.coroutines.reactive.awaitSingle
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController(
    private val testTlbRepository: TestTlbRepository,
) {

    companion object : KLogger()

    @GetMapping("/data-get")
    suspend fun findAll(): List<TestTlb> {
        log.info("${Thread.currentThread()}: message!!")
        return testTlbRepository.findAll().collectList().awaitSingle()
            .also {
                log.info("${Thread.currentThread()}: end before")
                delay(1000L)
                log.info("${Thread.currentThread()}: end")
            }
    }
}